temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowInternal.java

```diff
@@ -199,6 +188,73 @@ public static void registerListener(Object implementation) {
           .registerSignalHandlers(
               new WorkflowOutboundCallsInterceptor.RegisterSignalHandlersInput(requests));
     }
+
+    // Get all validators and lazily assign them to update handlers as we see them.
+    Map<String, POJOWorkflowMethodMetadata> validators = new HashMap<>();
```

Comment:
```suggestion
    Map<String, POJOWorkflowMethodMetadata> validators = new HashMap<>(workflowMetadata.getUpdateValidatorMethods().size());
```

Usually best to give a capacity to these collections where you know what they'll be. Especially short-lived ones like this. There are other collections in this review this same logic should apply to.